doc/sphinx/proofs/writing-proofs/reasoning-inductives.rst

```diff
@@ -284,155 +294,81 @@ Induction
-   instance, for an inductive type with  one constructor, the pattern notation
-   :n:`(p`:sub:`1` :n:`, ... , p`:sub:`n` :n:`)` can be used instead of
-   :n:`[ p`:sub:`1` :n:`... p`:sub:`n` :n:`]`.
+   .. exn:: Not an inductive product.
+      :undocumented:
 
-.. tacv:: induction @term with @bindings
+   .. exn:: Unable to find an instance for the variables @ident … @ident.
 
-   This behaves like :tacn:`induction` providing explicit instances for the
-   premises of the type of :n:`term` (see :ref:`bindings`).
+      Use the variant :tacn:`elim` `… with …` instead.
 
-.. tacv:: einduction @term
-   :name: einduction
+.. tacn:: einduction {+, @induction_clause } {? @induction_principle }
 
-   This tactic behaves like :tacn:`induction` except that it does not fail if
-   some dependent premise of the type of :n:`@term` is not inferable. Instead,
+   Behaves like :tacn:`induction` except that it does not fail if
+   some dependent premise of the type of :n:`@one_term` is not inferrable. Instead,
    the unresolved premises are posed as existential variables to be inferred
    later, in the same way as :tacn:`eapply` does.
 
-.. tacv:: induction @term using @term
-   :name: induction … using …
-
-   This behaves as :tacn:`induction`  but using :n:`@term` as induction scheme.
-   It does not expect the conclusion of the type of the first :n:`@term` to be
-   inductive.
-
-.. tacv:: induction @term using @term with @bindings
-
-   This behaves as :tacn:`induction … using …` but also providing instances
-   for the premises of the type of the second :n:`@term`.
-
-.. tacv:: induction {+, @term} using @qualid
-
-   This syntax is used for the case :n:`@qualid` denotes an induction principle
-   with complex predicates as the induction principles generated by
-   ``Function`` or ``Functional Scheme`` may be.
-
-.. tacv:: induction @term in @goal_occurrences
-
-   This syntax is used for selecting which occurrences of :n:`@term` the
-   induction has to be carried on. The :n:`in @goal_occurrences` clause is an
-   occurrence clause whose syntax and behavior is described in
-   :ref:`occurrences sets <occurrenceclauses>`. If variables or hypotheses not
-   mentioning :n:`@term` in their type are listed in :n:`@goal_occurrences`,
-   those are generalized as well in the statement to prove.
-
-   .. example::
-
-      .. coqtop:: reset all
-
-         Lemma comm x y : x + y = y + x.
-         induction y in x |-   *.
-         Show 2.
+.. tacn:: elim @one_term_with_bindings {? using @one_term {? with @bindings } }
 
-.. tacv:: induction @term with @bindings as @or_and_intropattern using @term with @bindings in @goal_occurrences
-          einduction @term with @bindings as @or_and_intropattern using @term with @bindings in @goal_occurrences
+   An older, more basic induction tactic.  We recommend using :tacn:`induction`
+   instead where possible.  :n:`@one_term` must have an inductive
```

Comment:
Interesting, that the argument of `induction` must have an inductive type was forgotten in the description of `induction`.

Reply:
I slipped it into the first bullet point.  Perhaps it should also be stated in "destruct"?

Reply:
It already appears in the description of `destruct`, although maybe not as clearly as it should:

>the inductive or co-inductive type in the one_term in induction_arg

"in the one_term" doesn't make it sufficiently clear that the type of one_term is the inductive (or co-inductive) type.

Reply:
A couple questions:
- For `A -> B -> C`, is it correct to say that A and B are premises and C is the conclusion?  This would be a second sense for "conclusion", which is currently defined as the statement to be proven.  Maybe say "the conclusion of a type" to clarify?
- Is it worth making "premise" a defined term? It seems fairly generic.  We could also say A and B are "antecedents".  That terminology is used twice in the doc with that meaning.
- Is there a proper term for `A -> B`?  An implication?

Reply:
>For `A -> B -> C`, is it correct to say that A and B are premises and C is the conclusion?  This would be a second sense for "conclusion", which is currently defined as the statement to be proven.  Maybe say "the conclusion of a type" to clarify?

Yes, this is correct. And yes, "conclusion" has this double meaning. I've signalled this when we've documented the other meaning (I was worried about the confusion that this would create).

>Is it worth making "premise" a defined term? It seems fairly generic. We could also say A and B are "antecedents". That terminology is used twice in the doc with that meaning.

Yes, I guess it would be worth (in the place where function types and dependent / non-dependent premises are explained). I am not used to seeing the terminology "antecedent" in this case, so I would avoid it.

>Is there a proper term for `A -> B`?  An implication?

Yes, an "implication" (with the propositional view) or a non-dependent function type (with the programming language / computational view).